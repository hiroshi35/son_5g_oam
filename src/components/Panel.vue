<template>
  <div :class="'panel ' + type">
    <div v-if="header" class="panel-heading">
      <slot name="header"></slot>
    </div>
    <div class="panel-body">
      <slot></slot>
    </div>
    <div v-if="footer" class="panel-footer">
      <slot name="footer"></slot>
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      type: {
        default: 'panel-white'
      },
      header: {
        type: Boolean,
        default: true
      },
      footer: {
        type: Boolean,
        default: false
      }
    }
  }
</script>