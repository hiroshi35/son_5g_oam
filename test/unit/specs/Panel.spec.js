import { shallow } from 'vue-test-utils'
import Panel from '@/components/Panel.vue'

describe('Panel.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Panel,
      {
        propsData: {
          type: 'panel-teal',
          footer: true,
          header: true
        },
        slots: {
          default: 'Hello World',
          header: 'Header',
          footer: 'Footer'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has a header with correct text', () => {
    expect(wrapper.find('.panel-heading').text()).toBe('Header')
    expect(wrapper.contains('.panel-heading')).toBeTruthy()
  })
  it('has a footer with correct text', () => {
    expect(wrapper.find('.panel-footer').text()).toBe('Footer')
    expect(wrapper.contains('.panel-footer')).toBeTruthy()
  })
  it('has a correct body text', () => {
    expect(wrapper.find('.panel-body').text()).toBe('Hello World')
  })
  it('it does not have a header when header: false', () => {
    wrapper.setProps({header: false})
    expect(wrapper.contains('.panel-heading')).toBeFalsy()
  })
  it('it does not have a footer when footer: false', () => {
    wrapper.setProps({footer: false})
    expect(wrapper.contains('.panel-footer')).toBeFalsy()
  })
  it('has correct props', () => {
    expect(wrapper.props().type).toBe('panel-teal')
    expect(wrapper.props().footer).toBeTruthy()
    expect(wrapper.props().header).toBeTruthy()
  })
  it('has correct class', () => {
    expect(wrapper.classes()).toContain('panel-teal')
  })
})
