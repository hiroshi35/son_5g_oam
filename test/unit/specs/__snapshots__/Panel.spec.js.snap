// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel.vue has the expected html structure 1`] = `
<div
  class="panel panel-teal"
>
  <div
    class="panel-heading"
  >
    Header
  </div>
   
  <div
    class="panel-body"
  >
    Hello World
  </div>
   
  <div
    class="panel-footer"
  >
    Footer
  </div>
</div>
`;
